### PR TITLE
feat(data): Implement GET /reading/count/device/name/{name} V2 API

### LINKED ISSUE #2961: [V2 data] Implement GET /reading/count/device/name/{name} V2 API
# 🚀 Feature Request

### Relevant Package

 This feature request is for core data

### Description

Per https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-data/2.x#/default/get_reading_count_device_name__name_, implement GET /reading/count/device/name/{name} V2 API.

### PR DESCRIPTION
**Pull request by judehung**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## Issue Number:
fix #2961 

## What is the new behavior?
Per https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-data/2.x#/default/get_reading_count_device_name__name_, implement GET /reading/count/device/name/{name} V2 API.

Also change method name from EventCountByDevice to
EventCountByDeviceName for consistency.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

File: internal/core/data/v2/application/reading.go [hunk 1]
```diff
@@ -49,3 +49,15 @@ func ReadingsByDeviceName(offset int, limit int, name string, dic *di.Container)
 	}
 	return readings, nil
 }
+
+// ReadingCountByDeviceName return the count of all of readings associated with given device and error if any
+func ReadingCountByDeviceName(deviceName string, dic *di.Container) (uint32, errors.EdgeX) {
+	dbClient := v2DataContainer.DBClientFrom(dic.Get)
```
Comment (lenny-intel) [inline internal/core/data/v2/application/reading.go:55]: Check DeviceName not empty. Same as all other query APIs

File: internal/core/data/v2/application/reading_test.go [hunk 2]
```diff
@@ -96,3 +96,19 @@ func TestReadingsByDeviceName(t *testing.T) {
 		})
 	}
 }
+
+func TestReadingCountByDeviceName(t *testing.T) {
```
Comment (lenny-intel) [inline internal/core/data/v2/application/reading_test.go:100]: Add test for when DeviceName is empty

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: internal/core/data/v2/application/reading.go (hunk 1 above)
Comment (judehung) [reply internal/core/data/v2/application/reading.go:58]: Added.

File: internal/core/data/v2/application/reading_test.go (hunk 2 above)
Comment (judehung) [reply internal/core/data/v2/application/reading_test.go:100]: Added.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Comment by judehung** [comment]
rebased

**Comment by judehung** [comment]
Fixed all occurrences of typo **416Example** in v2 core-data.yaml

**Comment by judehung** [comment]
rebased

**Review by cloudxxx8 — APPROVED** [review]
LGTM
